### PR TITLE
feat(pipelinetemplate): Metadata for UI; template protection

### PR DESCRIPTION
**Pull request by robzienert**
* Adds `protect` flag to templates, allowing template creators to prevent configuration-level modification of the stage graph. Going for the simplest level to start with, we can get more granular down the road if necessary.
* Adds new (optional) metadata block to templates containing `name`, `description` and `owner` so we can surface this in the UI for additional context.

@spinnaker/netflix-reviewers @spinnaker/google-reviewers PTAL

### CODE REVIEW

File: orca-pipelinetemplate/src/main/java/com/netflix/spinnaker/orca/pipelinetemplate/v1schema/TemplateMerge.java [hunk 1]
```diff
@@ -31,16 +31,21 @@ public static PipelineTemplate merge(List<PipelineTemplate> templates) {
...
 
       mergeConfiguration(result, template);
 
-      // TODO rz - might make sense that any child template must use inject syntax / modules.
       result.setStages(mergeIdentifiable(result.getStages(), template.getStages()));
-
       result.setModules(mergeIdentifiable(result.getModules(), template.getModules()));
     }
+
+    result.setMetadata(templates.get(templates.size() - 1).getMetadata());
```
Comment (jeyrschabu) [inline orca-pipelinetemplate/src/main/java/com/netflix/spinnaker/orca/pipelinetemplate/v1schema/TemplateMerge.java]: What is the story around this ?

**Comment by jeyrschabu** [comment]
One question. Other than that LGTM

**Comment by danielpeach** [comment]
It's still possible for config level parameters to mutate the graph, right? Mostly checking if I understand everything properly. e.g., if a parent template had:

```
id: manualJudgment
type: manualJudgment
inject:
  before: stage.{{variable}}
```

File: orca-pipelinetemplate/src/main/java/com/netflix/spinnaker/orca/pipelinetemplate/v1schema/TemplateMerge.java (hunk 1 above)
Comment (robzienert) [inline orca-pipelinetemplate/src/main/java/com/netflix/spinnaker/orca/pipelinetemplate/v1schema/TemplateMerge.java]: Will add a comment to the code, but we're only interested in applying the template metadata from the last template. Maybe this won't be necessary for the sake of building up pipeline executions, though.

**Comment by robzienert** [comment]
@danielpeach It isn't, no. I'm checking to see if any stages are defined as part of the configuration (what you pass in at exec/plan time). If any are defined, you'll get an error. Also, your example calls out that I need to update dcd-spec:

```
stages:
- id: manualJudgement
  type: manualJudgement
  inject:
    before: myStage <-- No "stage." or "module." namespacing anymore
```

Furthermore, `ConfigStageInjectionTransform` does not evaluate the templating language, so the injection rules are static.

**Pull request merged**